octopus-server/src/main/java/octopus/teamcity/server/OctopusBuildInformationBuildStartProcessor.java

```diff
@@ -40,6 +43,8 @@ public void updateParameters(final BuildStartContext buildStartContext) {
           }
           buildStartContext.addSharedParameter("octopus_vcstype", vcsType);
         }
+        final String buildUrl = webLinks.getViewLogUrl(buildStartContext.getBuild());
```

Comment:
This line is the crux of the whole change, it gets the URL that TeamCity says links to the build's UI page.